crates/ide_completion/src/snippet.rs

```diff
@@ -212,15 +212,14 @@ fn validate_snippet(
 ) -> Option<(Box<[GreenNode]>, String, Option<Box<str>>)> {
     let mut imports = Vec::with_capacity(requires.len());
     for path in requires.iter() {
-        let path = ast::Path::parse(path).ok()?;
-        let valid_use_path = path.segments().all(|seg| {
-            matches!(seg.kind(), Some(ast::PathSegmentKind::Name(_)))
-                || seg.generic_arg_list().is_none()
-        });
-        if !valid_use_path {
+        let use_path = ast::SourceFile::parse(&format!("use {};", path))
```

Comment:
cc @Veykril 

This looks iffy at first, but I think that's actually the right way to check things like "is this a valid import path?" -- just construct a file with import and check this directly!

Also, looking at this, what `||` intended to be `&&`?

Reply:
Ye that was supposed to be `&&`.

And I agree, this check is certainly nicer, it makes it more clear as well what is expected to be valid here.